### PR TITLE
Serverless incorrectly warns that TrustedKeyGroups as a CloudFront behavior is invalid 

### LINKED ISSUE #9595: Serverless incorrectly warns that `TrustedKeyGroups` as a CloudFront behavior is invalid
Serverless complains that `TrustedKeyGroups` isn't a valid attribute but it deploys correctly and CloudFormation documentation states that it's a valid part of the cache behaviour.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-cachebehavior.html#cfn-cloudfront-distribution-cachebehavior-trustedkeygroups

<details>
<summary><code>serverless.yml</code></summary>

```yaml
service: static
variablesResolutionMode: 20210326

provider:
  name: aws
  stage: ${opt:stage}
  versionFunctions: false
  runtime: nodejs12.x
  tracing:
    lambda: true
  region: us-east-1
  deploymentBucket:
    name: serverless-deploy
  stackTags:
    environment: ${self:provider.stage}
  logRetentionInDays: 30

resources:
  extensions:
    RouteLambdaFunction:
      DeletionPolicy: Delete
  Resources:
    CloudFrontDistribution:
      Type: AWS::CloudFront::Distribution
      Properties:
        DistributionConfig:
          Comment: !Join [ ' ', [ 'Otis Client stack', !Ref AWS::StackName ] ]
          DefaultCacheBehavior:
            CachePolicyId: 658327ea-f89d-4fab-a63d-7e88639e58f6
            OriginRequestPolicyId: 88a5eaf4-2fd4-4709-b370-b4c650ea3fcf
            TargetOriginId: s3/${self:custom.config.s3OriginBucketName}.s3.amazonaws.com${self:custom.config.s3OriginPath}
            TrustedKeyGroups:
              - ${self:custom.config.cloudFrontURLSigningKeyGroupId}
            ViewerProtocolPolicy: redirect-to-https
          Enabled: true
          Origins:
            - DomainName: ${self:custom.config.s3OriginBucketName}.s3.amazonaws.com
              Id: s3/${self:custom.config.s3OriginBucketName}.s3.amazonaws.com${self:custom.config.s3OriginPath}
              OriginPath: ${self:custom.config.s3OriginPath}
              S3OriginConfig:
                OriginAccessIdentity: origin-access-identity/cloudfront/${self:custom.config.cloudFrontOriginAccessIdentityId}
            - DomainName: ${self:custom.config.apiDomain}
              Id: custom/${self:custom.config.apiDomain}
              CustomOriginConfig:
                OriginProtocolPolicy: https-only
              
functions:
  route:
    handler: handlers/router.handler
    events:
      - cloudFront:
          eventType: viewer-request
          origin: s3://${self:custom.config.s3OriginBucketName}.s3.amazonaws.com${self:custom.config.s3OriginPath}
      - cloudFront:
          eventType: viewer-request
          pathPattern: /api/*
          origin: custom://${self:custom.config.apiDomain}
          behavior:
            ViewerProtocolPolicy: redirect-to-https
            AllowedMethods:
              - 'GET'
              - 'HEAD'
              - 'OPTIONS'
              - 'PUT'
              - 'POST'
              - 'PATCH'
              - 'DELETE'
            CachePolicyId: 4135ea2d-6df8-44a3-9df3-4b5a84be39ad
            TrustedKeyGroups:
              - ${self:custom.config.cloudFrontURLSigningKeyGroupId}

custom:
  config:
    cloudFrontURLSigningKeyGroupId: abcdef
    cloudFrontOriginAccessIdentityId: fedcba
    s3OriginBucketName: static
    s3OriginPath: /files
    apiDomain: api.serverless.com
```

</details>

<details>
<summary><b><code>serverless package --stage test</code> output</b></summary>

```
Serverless: Configuration warning at 'functions.route.events[1].cloudFront.behavior': unrecognized property 'TrustedKeyGroups'
Serverless:
Serverless: Learn more about configuration validation here: http://slss.io/configuration-validation
Serverless:
Serverless: Deprecation warning: Starting with next major, Serverless will throw on configuration errors by default. Adapt to this behavior now by adding "configValidationMode: error" to service configuration
            More Info: https://www.serverless.com/framework/docs/deprecations/#CONFIG_VALIDATION_MODE_DEFAULT
Serverless: Packaging service...
Serverless: Excluding development dependencies...
```

</details>

<b>Installed version</b>

```
Framework Core: 2.35.0 (local)
Plugin: 4.5.3
SDK: 4.2.2
Components: 3.8.3
```

### PR DESCRIPTION
**Pull request by Patil2099**
Fixes #9595

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/cloudFront.js [hunk 1]
```diff
@@ -163,6 +163,7 @@ class AwsCompileCloudFrontEvents {
         ViewerProtocolPolicy: {
           enum: ['allow-all', 'redirect-to-https', 'https-only'],
         },
+        TrustedKeyGroups:{type: 'array'}
```
Comment (pgrzesik) [inline lib/plugins/aws/package/compile/events/cloudFront.js:166]: Let's specify what type of items we expect in that array.

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thanks @Patil2099 :bow: I've added a comment, please let me know what do you think. Additionally, please make sure to run formatting via `npm run prettier` and `npm run lint` to ensure proper formatting, because I believe currently it will error out.

**Comment by pgrzesik** [comment]
Hello @Patil2099 :wave: Is there anything we could help you with to get this PR to the finish line?

**Comment by Patil2099** [comment]
Sorry, I forgot about the PR. Can you explain to me what comes inside the array? It would really helpful @pgrzesik.

**Comment by pgrzesik** [comment]
Hello @Patil2099 - I'm assuming it is based on https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-cachebehavior.html#cfn-cloudfront-distribution-cachebehavior-trustedkeygroups - so you'd need to verify what are the allowed values here and define schema for array items based on that.

File: lib/plugins/aws/package/compile/events/cloudFront.js (hunk 1 above)
Comment (preshetin) [reply lib/plugins/aws/package/compile/events/cloudFront.js:166]: I guess what @pgrzesik proposing in a change like this:

```suggestion
        TrustedKeyGroups:{type: 'array', items: {type: 'string'}}
```

So @Patil2099 feel free to add this change. If not... I'd be happy to finish it

**Comment by Patil2099** [comment]
Thanks for the commit Suggestion. I was about to do it today. I think it is ready now @pgrzesik

**Comment by pgrzesik** [comment]
Hello @Patil2099, thanks a lot for the improvement. It looks like the validation is failing - could you please look into that? In addition - is it possible to reference these groups e.g. with CF intrinsic functions such as `Ref`, etc? If yes, we should also allow schema to accept them.

**Comment by pgrzesik** [comment]
I'm going to close this one in favor of: #9884

**Pull request closed without merging**